src/components/Auth/SignupForm.tsx

```diff
@@ -1,33 +1,48 @@
 import PersonalInfoInput from 'components/common/PersonalInfoInput';
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 import styled from 'styled-components';
+import { theme } from 'styles/theme';
+import { UserSignupRequest } from 'types/user';
 
-function SignupForm() {
-  const [signupInfo, setSignupInfo] = useState({
-    name: '',
-    nickname: '',
-    email: '',
-    emailConfirm: '',
-    password: '',
-    passwordConfirm: '',
-  });
+interface SignupFormProps {
+  signupInfo: UserSignupRequest;
+  handleOnChange: (type: keyof UserSignupRequest, value: string) => void;
+  handleComplete: (type: keyof UserSignupRequest, value: boolean) => void;
+}
+
+function SignupForm(props: SignupFormProps) {
+  const { signupInfo, handleOnChange, handleComplete } = props;
+  const checkPassword = () => signupInfo.password.value !== signupInfo.passwordConfirm.value;
```

Comment:
```suggestion
  const isPasswordEqual = signupInfo.password.value === signupInfo.passwordConfirm.value;
```

이런 네이밍은 어떨까요?
checkPassword는 비밀번호가 같은가를 검사하는지 다른가를 검사하는지 불분명한것 같다는 생각이 들었어요.
결과적으로 handleComplete 함수에 complete 여부를 전달하는 것이니까 
```javascript
handleComplete('passwordConfirm', isPasswordEqual)
```
위와 같이 전달한다면, 
아 두 비밀번호 입력 값이 같을 때 완료되는구나고 바로 파악할 수 있다는 생각이 드네요!